frontend/rust-lib/flowy-database2/src/services/database_view/view_operation.rs

```diff
@@ -91,15 +91,15 @@ pub trait DatabaseViewOperation: Send + Sync + 'static {
 
   fn remove_calculation(&self, view_id: &str, calculation_id: &str);
 
-  fn get_all_filters(&self, view_id: &str) -> Vec<Arc<Filter>>;
+  fn get_all_filters(&self, view_id: &str) -> Vec<Filter>;
```

Comment:
using Arc then we don't need to clone the filter. The filter is cached in memory

Reply:
Or get_all_filters can just return &[Fitler]

Reply:
I haven't felt the need to clone the `Vec<Filter>`. I use this function to load the filters into the FilterController and it lives in a `RwLock<Vec<_>>` there. The other case is fetching the filters as PB to the frontend, which doesn't require shared access.

Reply:
Ok